tests/components/yardian/test_sensor.py

```diff
@@ -0,0 +1,156 @@
+async def test_sensor_entities(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    mock_config_entry: MockConfigEntry,
+    mock_yardian_client: object,
+) -> None:
+    """Snapshot all Yardian sensor entities."""
+
+    with patch("homeassistant.components.yardian.PLATFORMS", [Platform.SENSOR]):
+        await setup_integration(hass, mock_config_entry)
+
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+
+async def test_diagnostic_sensors(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_yardian_client: object,
+) -> None:
+    """Diagnostic sensors are disabled by default but report values when enabled."""
+
+    with patch("homeassistant.components.yardian.PLATFORMS", [Platform.SENSOR]):
+        await setup_integration(hass, mock_config_entry)
+
+    ent_reg = er.async_get(hass)
+
+    for uid in ("yid123_zone_delay", "yid123_water_hammer_duration"):
+        entity_id = ent_reg.async_get_entity_id("sensor", DOMAIN, uid)
+        assert entity_id is not None
+        reg_entry = ent_reg.async_get(entity_id)
+        assert reg_entry is not None
+        assert reg_entry.disabled
+        assert reg_entry.disabled_by is er.RegistryEntryDisabler.INTEGRATION
+
+
+def _set_oper_info(
+    coordinator: YardianUpdateCoordinator, updates: Mapping[str, object]
+) -> None:
+    """Update coordinator data with new oper_info values."""
+    device_state = coordinator.data
+    assert device_state is not None
+    oper_info_source = getattr(device_state, "oper_info", {})
+    try:
+        oper_info = dict(oper_info_source)
+    except TypeError:
+        oper_info = {}
+    oper_info.update(updates)
+    coordinator.async_set_updated_data(
+        YardianCombinedState(
+            zones=device_state.zones,
+            active_zones=device_state.active_zones,
+            oper_info=oper_info,
+        )
+    )
+
+
+async def _prepare_coordinator(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> YardianUpdateCoordinator:
+    """Set up the integration and return its coordinator."""
+    with patch("homeassistant.components.yardian.PLATFORMS", [Platform.SENSOR]):
+        await setup_integration(hass, mock_config_entry)
+    return hass.data[DOMAIN][mock_config_entry.entry_id]
+
+
+async def test_rain_delay_value_handles_invalid_data(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_yardian_client: object,
+) -> None:
+    """Rain delay sensor converts integers and ignores invalid data."""
+    coordinator = await _prepare_coordinator(hass, mock_config_entry)
+
+    assert _rain_delay_value(coordinator) == 3600
+
+    _set_oper_info(coordinator, {"iRainDelay": "bad-value"})
+    assert _rain_delay_value(coordinator) is None
+
+    _set_oper_info(coordinator, {"iRainDelay": None})
+    assert _rain_delay_value(coordinator) is None
```

Comment:
And for this test, to continue on my last comment, I don't really care that `_rain_delay_value` return a certain value, I want to see how that value is used. afaik it's a state, so have HA setup the integration and then just look at the state of the sensor instead.